### PR TITLE
DAOS-623 test: Improve run_utest.py

### PR DESCRIPTION
**Pull request by jolivier23**
Save stdout and stderr in logs
Print to output where to find logs for a failed test
Fix regression in run_go_test cmocka results (UTEST_control. prefix)

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by ryon-jensen — DISMISSED** [review]

**Comment by jolivier23** [comment]
@mjmac this restores old behavior in terms of categorizing the go tests under UTEST_control rather than (root)

File: Jenkinsfile [hunk 1]
```diff
@@ -798,7 +798,7 @@ pipeline {
                             // test results, and while code coverage is being
                             // added.
                             unitTestPost ignore_failure: true,
-                                         artifacts: ['covc_test_logs/*',
+                                         artifacts: ['covc_test_logs/',
```
[the remark this answers is not in the document]
Comment (jolivier23) [reply Jenkinsfile:801]: I'm not sure to be honest but this seems to work

File: utils/run_utest.py [hunk 2]
```diff
@@ -20,6 +20,9 @@
 import yaml
 
 
+HAS_CAPTURE = sys.version_info.major >= 3 and sys.version_info.minor >= 7
```
[the remark this answers is not in the document]
Comment (jolivier23) [reply utils/run_utest.py:23]: I'm not sure where it runs.  It worked on my Fedora system but failed in CI without this change.  I think it's 3.6 in CI

[the remark this answers is not in the document]
Comment (jolivier23) [reply utils/run_utest.py:23]: I guess according to Jenkinsfile, it's EL8

[the remark this answers is not in the document]
Comment (ashleypittman) [reply utils/run_utest.py:23]: el8 defaults to 3.6 but can have 3.10 I think.  pylint is checking for 3.6 so I guess that's good enough.

**Comment by jolivier23** [comment]
Ping reviewers

**Review by ashleypittman — APPROVED** [review]

**Review by ryon-jensen — APPROVED** [review]

**Pull request merged**